orchagent/dash/dashhaorch.cpp

```diff
@@ -0,0 +1,510 @@
+    {dash::types::HA_SCOPE_ROLE_SWITCHING_TO_ACTIVE, "HA_SCOPE_ROLE_SWITCHING_TO_ACTIVE"}
+};
+
+static std::map<dash::types::HaScope, std::string> ha_set_scope_to_str_map = {
+    {dash::types::SCOPE_UNSPECIFIED, "SCOPE_UNSPECIFIED"},
+    {dash::types::SCOPE_DPU, "SCOPE_DPU"},
+    {dash::types::SCOPE_ENI, "SCOPE_ENI"},
+};
+
+DashHaOrch::DashHaOrch(DBConnector *db, const vector<string> &tables, DashOrch *dash_orch, ZmqServer *zmqServer) :
+    ZmqOrch(db, tables, zmqServer),
+    m_dash_orch(dash_orch)
+{
+    SWSS_LOG_ENTER();
+}
+
+bool DashHaOrch::addHaSetEntry(const std::string &key, const dash::ha_set::HaSet &entry)
+{
+    SWSS_LOG_ENTER();
+
+    auto it = m_ha_set_entries.find(key);
+
+    if (it != m_ha_set_entries.end())
+    {
+        SWSS_LOG_WARN("HA Set entry already exists for %s", key.c_str());
+        return true;
+    }
+
+    uint32_t attr_count = 8;
+    sai_attribute_t ha_set_attr_list[8]={};
+    sai_status_t status;
+    sai_object_id_t sai_ha_set_oid = 0UL;
+
+    ha_set_attr_list[0].id = SAI_HA_SET_ATTR_LOCAL_IP;
+    ha_set_attr_list[0].value.ipaddr = covertPbIpaddrToSaiIpaddr(entry.local_ip());
+
+    ha_set_attr_list[1].id = SAI_HA_SET_ATTR_PEER_IP;
+    ha_set_attr_list[1].value.ipaddr = covertPbIpaddrToSaiIpaddr(entry.peer_ip());
+
+    ha_set_attr_list[2].id = SAI_HA_SET_ATTR_CP_DATA_CHANNEL_PORT;
+    ha_set_attr_list[2].value.u16 = static_cast<sai_uint16_t>(entry.cp_data_channel_port());
+    
+    ha_set_attr_list[3].id = SAI_HA_SET_ATTR_DP_CHANNEL_DST_PORT;
+    ha_set_attr_list[3].value.u16 = static_cast<sai_uint16_t>(entry.dp_channel_dst_port());
+
+    ha_set_attr_list[4].id = SAI_HA_SET_ATTR_DP_CHANNEL_MIN_SRC_PORT;
+    ha_set_attr_list[4].value.u16 = static_cast<sai_uint16_t>(entry.dp_channel_src_port_min());
+
+    ha_set_attr_list[5].id = SAI_HA_SET_ATTR_DP_CHANNEL_MAX_SRC_PORT;
+    ha_set_attr_list[5].value.u16 = static_cast<sai_uint16_t>(entry.dp_channel_src_port_max());
+
+    ha_set_attr_list[6].id = SAI_HA_SET_ATTR_DP_CHANNEL_PROBE_INTERVAL_MS;
+    ha_set_attr_list[6].value.u32 = entry.dp_channel_probe_interval_ms();
+
+    ha_set_attr_list[7].id = SAI_HA_SET_ATTR_DP_CHANNEL_PROBE_FAIL_THRESHOLD;
+    ha_set_attr_list[7].value.u32 = entry.dp_channel_probe_fail_threshold();
+
+    status = sai_dash_ha_api->create_ha_set(&sai_ha_set_oid,
+                                                gSwitchId,
+                                                attr_count,
+                                                ha_set_attr_list);
+
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_ERROR("Failed to create HA Set object in SAI for %s", key.c_str());
+        task_process_status handle_status = handleSaiCreateStatus((sai_api_t) SAI_API_DASH_HA, status);
+        if (handle_status != task_success)
+        {
+            return parseHandleSaiStatusFailure(handle_status);
+        }
+    }
+    m_ha_set_entries[key] = HaSetEntry {sai_ha_set_oid, entry};
```

Comment:
What's the expected scale/how many entries do we expect per DPU?

Reply:
1 ha set per DPU.